src/Uno.UWP/Media/Playback/MediaPlayer.Android.cs

```diff
@@ -0,0 +1,474 @@
+using System;
+using Android.App;
+using Android.Widget;
+using Uno.Media.Playback;
+using Windows.Media.Core;
+using Android.Media;
+using Android.OS;
+using Java.Lang;
+using Java.Util.Concurrent;
+using Uno.Extensions;
+using Android.Content.Res;
+using System.IO;
+using Android.Content;
+using Android.Views;
+using Android.Graphics;
+using Android.Runtime;
+using Uno.Logging;
+using AndroidMediaPlayer = Android.Media.MediaPlayer;
+
+namespace Windows.Media.Playback
+{
+	public partial class MediaPlayer :
+		Java.Lang.Object,
+		ISurfaceHolderCallback,
+		AndroidMediaPlayer.IOnCompletionListener,
+		AndroidMediaPlayer.IOnErrorListener,
+		AndroidMediaPlayer.IOnPreparedListener,
+		AndroidMediaPlayer.IOnSeekCompleteListener,
+		AndroidMediaPlayer.IOnBufferingUpdateListener,
+		AndroidMediaPlayer.IOnVideoSizeChangedListener,
+		View.IOnLayoutChangeListener
+	{
+		private AndroidMediaPlayer _player;
+
+		private bool _isPlayRequested = false;
+		private bool _isPlayerPrepared = false;
+		private bool _hasValidHolder = false;
+		private VideoStretch _currentStretch = VideoStretch.Uniform;
+		private bool _isUpdatingStretch = false;
+
+		private IScheduledExecutorService _executorService = Executors.NewSingleThreadScheduledExecutor();
+		private IScheduledFuture _scheduledFuture;
+		private AudioPlayerBroadcastReceiver _noisyAudioStreamReceiver;
+
+		const string MsAppXScheme = "ms-appx";
+		const string MsAppDataScheme = "ms-appdata";
+
+		public virtual IVideoSurface RenderSurface { get; private set; } = new VideoSurface(Application.Context);
+
+		private void Initialize()
+		{
+			((VideoSurface)RenderSurface).AddOnLayoutChangeListener(this);
+
+			// Register intent to pause media when audio become noisy (unplugged headphones, for example) 
+			_noisyAudioStreamReceiver = new AudioPlayerBroadcastReceiver(this);
+			var intentFilter = new IntentFilter(AudioManager.ActionAudioBecomingNoisy);
+			Application.Context.RegisterReceiver(_noisyAudioStreamReceiver, intentFilter);
```

Comment:
Is this ever unregistered?

Reply:
Call to Unregistered added when player is disposed